lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -255,6 +255,9 @@ && isIntegerOrDecimalPrimitive(primitiveType)) {
                 }
```

Comment:
there is still a kill switch, right?

Reply:
yes, the flag `parquet.optimized-reader.enabled` is still there, this just avoids silent fallbacks to old reader because now all the types are expected to be supported.